hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/services/TestExponentialRetryPolicy.java

```diff
@@ -67,6 +87,130 @@ public void testDefaultMaxIORetryCount() throws Exception {
+    AbfsClientThrottlingIntercept instance3 = AbfsClientThrottlingInterceptFactory.getInstance(accountName, configuration);
+    AbfsClientThrottlingIntercept instance4 = AbfsClientThrottlingInterceptFactory.getInstance(accountName1, configuration);
+    AbfsClientThrottlingIntercept instance5 = AbfsClientThrottlingInterceptFactory.getInstance(accountName, configuration);
+    //if singleton is not enabled, for different accounts instances should return different value
+    assertNotEquals(instance3, instance4);
+
+    //if singleton is not enabled, for same accounts instances should return same value
+    assertEquals(instance3, instance5);
+  }
+
+  @Test
+  public void testOperationOnAccountIdle() throws Exception {
+    //Get the filesystem.
+    AzureBlobFileSystem fs = getFileSystem();
+    AbfsConfiguration configuration1 = fs.getAbfsStore().getClient().getAbfsConfiguration();
+    Assume.assumeTrue(configuration1.isAccountThrottlingEnabled());
+
+    AbfsClientThrottlingIntercept accountIntercept = AbfsClientThrottlingInterceptFactory.getInstance(this.getAccountName(),
+            configuration1);
+    int bufferSize = MIN_BUFFER_SIZE;
+    final byte[] b = new byte[2 * bufferSize];
+    new Random().nextBytes(b);
+
+    Path testPath = path(TEST_PATH);
+
+    //Do an operation on the filesystem.
+    FSDataOutputStream stream = fs.create(testPath);
+
+    try {
+      stream.write(b);
+    } finally{
+      stream.close();
+    }
+
+    //Don't perform any operation on the account.
+    int sleepTime = (int)((getAbfsConfig().getAccountOperationIdleTimeout()) * 1.5);
+    Thread.sleep(sleepTime);
+
+    FSDataInputStream streamRead = fs.open(testPath);
+    try {
+      streamRead.read(b);
+    } finally{
+      streamRead.close();
+    }
+
+    //Perform operations on another account.
+    AzureBlobFileSystem fs1 = new AzureBlobFileSystem();
+    Configuration config = new Configuration(getRawConfiguration());
+    String accountName1 = config.get(FS_AZURE_ABFS_ACCOUNT1_NAME);
+    final String abfsUrl1 = this.getFileSystemName() + "12" + "@" + accountName1;
+    URI defaultUri1 = null;
+    try {
+      defaultUri1 = new URI("abfss", abfsUrl1, null, null, null);
+    } catch (Exception ex) {
+      throw new AssertionError(ex);
+    }
+    fs1.initialize(defaultUri1, getRawConfiguration());
+    AbfsConfiguration configuration2 = fs1.getAbfsStore().getClient().getAbfsConfiguration();
+    AbfsClientThrottlingIntercept accountIntercept1 = AbfsClientThrottlingInterceptFactory.getInstance(accountName1,
+            configuration2);
+    FSDataOutputStream stream1 = fs1.create(testPath);
+    try {
+      stream1.write(b);
+    } finally{
+      stream1.close();
+    }
+
+    //Verfiy the write analyzer for first account is idle but the read analyzer is not idle.
+    if (accountIntercept != null){
+      Assert.assertTrue(accountIntercept.getWriteThrottler().getIsOperationOnAccountIdle().get());
```

Comment:
add a mesage on all these asserts. think "if this assert failed on a jenkins run, what would i need to know?"